server/py/services/alerts/crud/alerts.py

```diff
@@ -421,7 +421,10 @@ def reset_alert(self, session: sqlalchemy.orm.Session, project: str, name: str):
                 session, alert.id
             )
             activation_time = alert_state.get("last_updated")
-            activation_id = alert_state.get("full_object", {}).get("last_activation_id")
+            # or {} is needed of the case if full_object is None
+            activation_id = (alert_state.get("full_object") or {}).get(
+                "last_activation_id"
+            )
```

Comment:
if `alert_state` is None then getting `"last_updated"` will also raise the same exception.
You can instead do:
```python
            alert_state = alert_state or {}
            activation_time = alert_state.get("last_updated")
            activation_id = alert_state.get("full_object", {}).get("last_activation_id")
```

Reply:
@TomerShor the problem here is not that alert_state is empty but `alert_state["full_object""]` is None. 
This string `activation_id = alert_state.get("full_object", {}).get("last_activation_id")` you suggested is something we better avoid (see description of PR why).

Regarding empty `alert_state`, technically it's not possible to get `None` here as we use `.one()` expression https://github.com/mlrun/mlrun/blob/06c5da78dc4f539f85f9a4466387ee86b99f909a/server/py/framework/db/sqldb/db.py#L5925 to get record from db. But I agree that it's better to be on safe side, so I'll add a warning.

Reply:
It is confusing because looking at `get_alert_state_dict` it seems that you can get None, but I see why this is not the case.
😓